### ISSUE TITLE
suitcase-zarr

### ISSUE THREAD

**Issue by ke-zhang-rd** [bluesky/suitcase-core#74]
This issue is a pre-PR discussion about `suitcase-zarr`.

`zarr` document https://zarr.readthedocs.io/en/stable/index.html
`suitcase-zarr` need follow https://blueskyproject.io/suitcase/writing.html
[documents_to_xarray ](https://github.com/bluesky/intake-bluesky/blob/913039a4067bdc211b27639af9692c7d0b40e787/intake_bluesky/core.py#L123) functioin in `intake-bluesky` could be a reference of design pattern

**Comment by danielballan** [comment]
It would be great if `intake-bluesky` would provide a simple function that takes in a generator of `(name, doc)` pairs and returns `xarray.Dataset` objects. Specifically, each _stream_ can be represented as one `xarray.Dataset`, so perhaps it makes sense to return a dict mapping each stream name to an `xarray.Dataset`.

Currently, `intake-bluesky` doesn't have a function like that. It has `documents_to_xarray` which takes in a bunch of callables. That aspect of `intake-bluesky` is under active development by @gwbischof. In the meantime, here is a somewhat convoluted approach that should work. We can replace it with a simpler approach as soon as one is available.

```python
from intake_bluesky.in_memory import BlueskyInMemoryCatalog

def documents_to_xarrays(documents):
    "Consume a generator instance of (name, doc) pairs. Return dict of xarray.Datasets."

    def gen():
        # single-use generator function wrapping a generator instance
        yield from documents

    cat = BlueskyInMemoryCatalog()
    cat.upsert(gen, (), {})
    xarrays = {stream_name: cat[stream_name].read() for stream_name in cat}
    return xarrays
```

**Comment by ke-zhang-rd** [comment]
@danielballan 
I just got some update from garrett. Looks there will be a function `read` of `BlueskyEventStream` which later use multiple generator of `start`, `stop` and `event` as argument. So I could just prepare those generator in Serializer for `read` and get `xarray` back and do `xarray.to_zarr`. But this require `intake-bluesky` as dependency.

**Comment by danielballan** [comment]
Yes. I like that plan. I think it is OK to add `intake-bluesky` as a dependency for the the zarr suitcase. The alternatives are:

* Keep a separate copy of the the documents-to-xarray code. This means we'd have to remember to keep it in sync.
* Pull the documents-to-xarray code into a new package that intake-bluesky and suitcase-zarr both import. Since intake-bluesky and suitcase are already _very_ modularized, I would rather not split them up any further.

So I think this is better than the alternatives.
